### ISSUE TITLE
vscode crouton chroots reset the chrome book 

### ISSUE THREAD

**Issue by TomTravis** [dnschneid/crouton#4747]
```vscode crashes and every chroot i have tried Xorg and Xiwi

crostini runs it just fine

Here the Instructions and the link, skip part 1 of you just clicked on the link and downloaded it.

Step 1; curl -L https://aka.ms/linux-arm64-deb > code_arm64.deb

Step 2; sudo apt install ./code_arm64.deb
![Screenshot 2022-05-23 2 28 18 PM](https://user-images.githubusercontent.com/46386261/169909440-36fb1f20-f58c-4131-8a70-c26ed3f4eae3.png)

Please paste the output of the following command here: sudo edit-chroot -all
```

**Comment by TomTravis** [comment]
it really does not matter, it kills the xserver and reset the chorme book 
jammy bullseye and my own debootstrap 
The latest Xerver is now running
![Screenshot_2022-05-24_15-51-11](https://user-images.githubusercontent.com/46386261/170145029-c5493d1a-0b78-4f57-ab72-c553d9b37618.png)
[x-help.txt](https://github.com/dnschneid/crouton/files/8766697/x-help.txt)
[xrdb.txt](https://github.com/dnschneid/crouton/files/8766699/xrdb.txt)

Xserver is working again stable no reset Xresources can be local the performance is great, xiwi has issues and needs work
I will update me fork this week lots of new work done

**Comment by TomTravis** [comment]
I can tell you no logs are made the debug status is unknown, vscode init starts fails and the chrome book resets.
vscode create a dirty file system on exit. You must umount, fsck the FS and then enter chroot again, if you do not clean the FS the chromebook just resets on chroot entry. vscode is now removed from my chroot the git repo is installed still with yarn and npm nodejs also this is needed to chromeOS build nodejs, You are required to do pull request to keep current on vscode for build, yarn is a verbose build app, it is strange to work with. Many versions and it gets confused. nodejs yarn and npm.

maybe a Azure Repo could be better i have yet to attempt a build on github arm64 builds maybe

**Comment by BetaPictoris** [comment]
Are there not any Visual Studio Code logs in `~/.config/Code/logs`? Does the problem still persist if you use another Electron application, like Slack, Atom, or Notion? **It still helps if you give the output of `sudo edit-chroot -all` or `uname -a`**.

**Comment by TomTravis** [comment]
no logs,vscode never starts, it causes many FS errors
there is no config.code it never gets created

Electron application, like Slack, Atom, or Notion? have not tried this?
i use gedit with plugins for low lever work

this one of the CR
chronos@localhost /media/removable/sd200 $ cd bull
chronos@localhost /media/removable/sd200/bull $ sudo edit-chroot -c /media/removable/sd200/bull/chroots -all
name: bullseye
encrypted: no
Entering /media/removable/sd200/bull/chroots/bullseye...
crouton: version 1-20220512214808
release: bullseye
architecture: arm64
xmethod: xorg
targets: audio,xiwi,xorg
host: version 14588.123.0 (Official Build) stable-channel jacuzzi 
kernel: Linux localhost 5.10.106-15290-g6e9d70c8a9b4 #1 SMP PREEMPT Wed May 18 19:34:18 PDT 2022 aarch64 GNU/Linux
freon: yes
Unmounting /media/removable/sd200/bull/chroots/bullseye...
chronos@localhost /media/removable/sd200/bull $ 

name: bullseye
encrypted: no
Entering /media/removable/sd200/deb/chroots/bullseye...
crouton: version 1-20220512214808
release: bullseye
architecture: arm64
xmethod: xorg
targets: xorg,xiwi,core,audio
host: version 14588.123.0 (Official Build) stable-channel jacuzzi 
kernel: Linux localhost 5.10.106-15290-g6e9d70c8a9b4 #1 SMP PREEMPT Wed May 18 19:34:18 PDT 2022 aarch64 GNU/Linux
freon: yes
Unmounting /media/removable/sd200/deb/chroots/bullseye.

**Comment by TomTravis** [comment]
the vscode  issue was submitted the robot asked to install the latest version of vscode it was 1.67.2

**Comment by TomTravis** [comment]
yes good
hopeful

**Comment by TomTravis** [comment]
the electron build is unknown. it[electron](https://www.electronjs.org/docs/latest/development/build-instructions-linux#prerequisites)
I may have to build on amd64 and cross compile the src has been downloading for 8hr  gclient takes forever

 Electron puts Mac OS and Linux together not so bad arm64 Debian build may have issues

running cli tests did not reset from npm binaries, running .src/scripts/test.sh did reset the machine.
the ChromeOS VScode was likely built for it to run,

**Comment by BetaPictoris** [comment]
> the electron build is unknown. it [electron](https://www.electronjs.org/docs/latest/development/build-instructions-linux#prerequisites) I may have to build on amd64 and cross compile the src has been downloading for 8hr gclient takes forever
> 
> Electron puts Mac OS and Linux together not so bad arm64 Debian build may have issues
> 
> running cli tests did not reset from npm binaries, running .src/scripts/test.sh did reset the machine. the ChromeOS VScode was likely built for it to run,

I'm sorry if I don't really understand what you are trying to get at here but, are you attempting you build everything from source? If that is the case I don't think there is that much we can do to help you.

**Comment by TomTravis** [comment]
yes VScode has it unique Electron node_module, the proposed it has nothing to do with Vscode and electron is at fault is complete BS



> [<img alt="@BetaPictoris" width="20" height="20" src="https://avatars.githubusercontent.com/u/65696362?s=60&amp;u=e679424275321cc416c3b2165ff1240337998dbf&amp;v=4">](/BetaPictoris) [BetaPictoris](/BetaPictoris) mentioned this issue [ 3 days ago ](#ref-issue-1250253259)
> [ vscode X11 chroot on ChromeOS, init fail reset HW microsoft/vscode#150500](/microsoft/vscode/issues/150500)

**Comment by TomTravis** [comment]
> > the electron build is unknown. it [electron](https://www.electronjs.org/docs/latest/development/build-instructions-linux#prerequisites) I may have to build on amd64 and cross compile the src has been downloading for 8hr gclient takes forever
> > Electron puts Mac OS and Linux together not so bad arm64 Debian build may have issues
> > running cli tests did not reset from npm binaries, running .src/scripts/test.sh did reset the machine. the ChromeOS VScode was likely built for it to run,
> 
> I'm sorry if I don't really understand what you are trying to get at here but, are you attempting you build everything from source? If that is the case I don't think there is that much we can do to help you.

the test now run after days of saying iIm spamming microsoft, issues out of scope some folks are helpful
arm64 still fails x86_64 builds
![Screenshot at 2022-05-31 12-57-33](https://user-images.githubusercontent.com/46386261/171275554-4feb7ff8-8ad0-43a1-bbff-0d81a916a40f.png)
[npm-sh.log](https://github.com/dnschneid/crouton/files/8809045/npm-sh.log)

**Comment by TomTravis** [comment]
![Screenshot at 2022-05-31 19-43-01](https://user-images.githubusercontent.com/46386261/171317590-edb2e4da-e57c-4c8d-9e42-f1424ec16fb6.png)

**Comment by BetaPictoris** [comment]
> yes VScode has it unique Electron node_module, the proposed it has nothing to do with Vscode and electron is at fault is complete BS
> 
> > [<img alt="@BetaPictoris" width="20" height="20" src="https://avatars.githubusercontent.com/u/65696362?s=60&amp;u=e679424275321cc416c3b2165ff1240337998dbf&amp;v=4">](/BetaPictoris) [BetaPictoris](/BetaPictoris) mentioned this issue [ 3 days ago ](#ref-issue-1250253259)
> > [ vscode X11 chroot on ChromeOS, init fail reset HW microsoft/vscode#150500](/microsoft/vscode/issues/150500)

It was an idea, and you didn't give us any logs or error messages to work off of! You also didn't say if you even tested other Electron applications. For all I know it could've been an issue with Electron.

**Comment by TomTravis** [comment]
im laughing now 

> Electron puts Mac OS and Linux together not so bad arm64 Debian build may have issues
> running cli tests did not reset from npm binaries, running .src/scripts/test.sh did reset the machine. the ChromeOS VScode was likely built for it to run,

maybe you not remember [A/UX](https://en.wikipedia.org/wiki/A/UX) and apple, or [NeXT](https://en.wikipedia.org/wiki/NeXT) systems and small talk, the [Mac OS](https://en.wikipedia.org/wiki/MacOS) is linux made to run Mac Binaries

this was a long time ago, [SUN microsytem](https://en.wikipedia.org/wiki/Sun_Microsystems) still was building computers Bought by Oracle.

 Like chromeOS and android it will come together.

There is no log or error messages none, this was the challenge, Binaries made to Mac arm64 (M1) do not run on Debian Linux this was the reset, The arm64 is still not working i may not work. I have compiled Node for arm64 now vscode requires x64 node for some reason, Chromium is the same, cross compile is the only option

["IBM AiX Microsoft Linux"](https://en.wikipedia.org/wiki/IBM_AIX)
["DEC Ultrix"](https://en.wikipedia.org/wiki/Ultrix)

Chromuim now reset the chromebook just like vscode did, no logs now error messages
vscode make the crouton src tree easy to look at now work can happen
lot of work to do now, Android Studio is next
